composer.json

```diff
@@ -23,18 +23,21 @@
         }
     ],
     "require": {
-        "php": ">=5.3.3",
+        "php": ">=5.5",
         "zendframework/zend-modulemanager": "2.*",
-        "zendframework/zend-stdlib": "2.*",
-        "zendframework/zend-servicemanager": "2.*",
+        "zendframework/zend-stdlib": "2.* || 3.*",
+        "zendframework/zend-servicemanager": "^2.7.5 || 3.*",
         "zendframework/zend-loader": "2.*",
-        "zendframework/zend-eventmanager": "2.*",
-        "zendframework/zend-mvc": "2.*",
+        "zendframework/zend-eventmanager": "2.* || 3.*",
+        "zendframework/zend-mvc": "2.* || 3.*",
         "kriswallsmith/assetic": "~1.2.1"
     },
     "require-dev": {
-        "zendframework/zendframework": "2.*",
-        "phpunit/phpunit": "~4.0",
+        "zendframework/zend-http": "2.*",
+        "zendframework/zend-view": "2.*",
+        "zendframework/zend-cache": "2.*",
+        "zendframework/zend-console": "2.*",
+        "phpunit/phpunit": "~4.8",
         "squizlabs/php_codesniffer": "1.5.*"
     },
```

Comment:
I'd recommend the following:

```json
    "require": {
        "php": "^5.5 || ^7.0",
        "zendframework/zend-modulemanager": "^2.7.2",
        "zendframework/zend-stdlib": "^2.7.7 || ^3.0.1",
        "zendframework/zend-servicemanager": "^2.7.6 || ^3.1.1",
        "zendframework/zend-loader": "^2.5.1",
        "zendframework/zend-eventmanager": "^2.6.3 || ^3.0.1",
        "zendframework/zend-mvc": "^2.7.9 || ^3.0.2",
        "kriswallsmith/assetic": "~1.2.1"
    },
    "require-dev": {
        "zendframework/zend-http": "^2.5.4",
        "zendframework/zend-view": "^2.8.1",
        "zendframework/zend-cache": "^2.7.1",
        "zendframework/zend-console": "^2.6",
        "phpunit/phpunit": "^4.8",
        "squizlabs/php_codesniffer": "~1.5.0"
    }
```

The above constraints ensure that it's pinned to versions that are known to be forwards-compatible with v3 releases; these have also been tested rigorously to ensure they remain BC with v2 releases.